### ISSUE TITLE
TypeError: cannot concatenate 'str' and 'list' objects

### ISSUE THREAD

**Issue by rsaeks** [chrismarget/certbot-asa#9]
When attempting to renew a certificate, the above error message is displayed and the certificate does not renew.

**Comment by chrismarget** [comment]
I have a theory (argparser returns varying types depending on object count) about this error, but it's pretty much moot now:

https://community.letsencrypt.org/t/2018-01-09-issue-with-tls-sni-01-and-shared-hosting-infrastructure/49996

**Comment by rsaeks** [comment]
I feel this may not be related to the linked issue as this has been occurring in the past:

https://github.com/chrismarget/certbot-asa/issues/3

**Comment by chrismarget** [comment]
Yes, it's the same issue. My point is, the plugin is dead. With LetsEncrypt no longer supporting TLS-SNI-01, there's nothing to fix.

**Comment by rsaeks** [comment]
The plugin being certbot-asa or another plugin?

**Comment by chrismarget** [comment]
This plugin. certbot-asa. It's dead.

**Comment by d3pr3cat3d** [comment]
Any plans to update this to update so that TLS-SNI-01 is not used?

**Comment by chrismarget** [comment]
> Any plans to update this to update this so that TLS-SNI-01 is not used?

The alternatives are HTTP and DNS challenges.

I don't think an ASA can respond to an HTTP challenge - If I'm mistaken about this I'll be delighted to update the plugin so it can answer by HTTP.

I do not plan to pursue the DNS challenge.
